astroquery/splatalogue/core.py

```diff
@@ -76,14 +76,14 @@ def set_default_options(self, **kwargs):
         """
         self.data.update(self._parse_kwargs(**kwargs))
 
-    def get_species_ids(self, *, restr=None, reflags=0, recache=False):
+    def get_species_ids(self, regex_str=None, *, reflags=0, recache=False):
         """
         Get a dictionary of "species" IDs, where species refers to the molecule
         name, mass, and chemical composition.
 
         Parameters
         ----------
-        restr : str
+        regex_str : str
             String to compile into an re, if specified.   Searches table for
```

Comment:
it's not an implementation detail exactly, but yes, this could be rephrased.  It's important that the user know that the string is intended as a python `re` regular expression, since `re` languages differ between every implementation.